pkg/service/utgen/ai.go

```diff
@@ -107,6 +109,25 @@ func (ai *AIClient) Call(ctx context.Context, prompt *Prompt, maxTokens int) (st
 
 	var apiBaseURL string
 
+	// Signal handling for Ctrl+C (SIGINT)
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+	defer signal.Stop(sigChan)
+
+	// Create a cancellable context for the request
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Start a goroutine to listen for Ctrl+C and cancel the context
```

Comment:
creating a fresh context won't be a good idea..we try to use same context to different level so that same context cancellation and handler can stop entire thing at a point